specification/cognitiveservices/data-plane/FormRecognizer/preview/v2.0/FormRecognizer.json

```diff
@@ -0,0 +1,1524 @@
+          "type": "object",
+          "x-nullable": true,
+          "properties": { 
+            "count": {
+              "description": "Current count of trained models.",
+              "type":"integer",
+              "x-nullable": false
+            },
+            "limit": {
+              "description": "Max number of models that can be trained for this subscription.",
+              "type": "integer",
+              "x-nullable": false
+            },
+            "lastUpdatedDateTime": {
+              "format": "date-time",
+              "description": "Get or set the summary last updated datetime.",
+              "type": "string"
+            }
+          }
+        },        
+        "modelList": {
+          "description": "Collection of models.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ModelInfo"
+          }
+        },
+        "nextLink": {
+          "description": "Next link",
+          "type": "string"
+        }
+      }
+    },
+    "Model": {
+      "description": "Result of a model status query operation.",
+      "type": "object",
+      "required": [
+        "modelInfo"
+      ],
+      "properties": {
+        "modelInfo":{
+          "$ref": "#/definitions/ModelInfo",
+          "description":"Basic Model info"
+        },
+        "keys": {
+          "$ref": "#/definitions/KeysResult",
+          "description": "Keys extracted by model."
+        },
+        "trainResult": {
+          "description": "Train batch operation result.",
+          "$ref": "#/definitions/TrainResult"
+        }        
+      }
+    },
+    "KeysResult": {
+      "description": "Keys extracted by a model.",
+      "type": "object",
+      "required": [
+        "clusters"
+      ],
+      "properties": {
+        "clusters": {
+          "description": "Object mapping ClusterIds to Key lists.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "FormDocumentReport": {
+      "type": "object",
+      "required": [
+        "documentName",
+        "pages",
+        "errors",
+        "status"
+      ],
+      "properties": {
+        "documentName": {
+          "description": "Reference to the data that the report is for.",
+          "type": "string"
+        },
+        "pages": {
+          "format": "int32",
+          "description": "Total number of pages trained on.",
+          "type": "integer",
+          "x-nullable": false
+        },
+        "errors": {
+          "description": "List of errors per page.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "status": {
+          "description": "Status of the training operation.",
+          "enum": [
+            "succeeded",
+            "partiallySucceeded",
```

Comment:
What does "partiallySucceeded" mean? In GET /custom/models/{modelId}, if model status is "invalid", would there still be training operation status? Can the training operation status be "succeeded" while the model status be "invalid"?

Reply:
The global training operation status is reported under modelInfo->status ('creating', 'ready', 'invalid'). The status reported under trainResult->trainingDocuments[idx]->status is the status ('succeeded', 'partiallySucceeded', 'failed') of a single training document. You could have 10 training documents with 5 pages each and an error could be reported on page 3 of document 7. Then the status for that particular document would be 'partiallySucceeded'. This does not prevent the overall modelInfo->status to be 'ready'. Similarly, you could have all the training documents have a 'succeeded' status, but report some fatal training error in trainResult->errors. In that case, the global modelInfo->status would be 'invalid'.

Reply:
I'm worried this would be a documentation nightmare. Do customers really need such detailed status report? Or we have it because we want to use it for debugging somehow?